app/src/main/java/com/hedvig/app/feature/marketpicker/MarketSelectedFragment.kt

```diff
@@ -47,10 +48,18 @@ class MarketSelectedFragment : Fragment(R.layout.fragment_market_selected) {
                     setImageDrawable(context.compatDrawable(market.flag))
                 }
                 setHapticClickListener {
-                    viewModel.navigateTo(
-                        CurrentFragment.MARKET_PICKER,
-                        signUp to "marketButton"
-                    )
+                    val transaction = parentFragmentManager.beginTransaction()
```

Comment:
Are we mixing navigation/back stack management strategies in this screen now? 🤔

Reply:
You are right, should navigate via view model here as well, fixing!